ifrs17/Report/ReportStorage.ipynb

```diff
@@ -138,6 +147,12 @@
                 "\n        await hierarchicalDimensionCache.InitializeAsync<AmountType>();  ",
                 "\n        await hierarchicalDimensionCache.InitializeAsync<VariableType>();",
                 "\n        await hierarchicalDimensionCache.InitializeAsync<ReportingNode>();",
+                "\n",
+                "\n        // Initial Values",
+                "\n        var mostRecentPartition = (await workspace.Query<PartitionByReportingNodeAndPeriod>().Where(x => x.Scenario == null).OrderBy(x => x.Year).ThenBy(x => x.Month).ToArrayAsync()).Last();",
```

Comment:
Maybe LastOrDefault, and default to current Year and Month.

Also maybe you can select the Year and Month in the DB already, instead of loading the full entity.

Reply:
Same for the reporting node.

Reply:
Changed it to LastOrDefault() and FirstOrDefault().